File: src/spice2x/api/resources/dart/spiceapi/src/wrappers/analogs.dart
part of spiceapi;

class AnalogState {
  String name;
  double state;
  bool active;

  AnalogState(this.name, this.state);
  AnalogState._fromRead(this.name, this.state, this.active);
}

Future<List<AnalogState>> analogsRead(Connection con) {
  var req = Request("analogs", "read");
  return con.request(req).then((res) {

    // build states list
    List<AnalogState> states = [];
    for (List state in res.getData()) {
      states.add(AnalogState._fromRead(
        state[0],
        state[1],
        state[2],
      ));
    }

    // return it
    return states;
  });
}

Future<void> analogsWrite(Connection con, List<AnalogState> states) {
  var req = Request("analogs", "write");

  // add params
  for (var state in states) {
    var obj = [
      state.name,
      state.state
    ];
    req.addParam(obj);
  }

  return con.request(req);
}

Future<void> analogsWriteReset(Connection con, List<String> names) {
  var req = Request("analogs", "write_reset");

  // add params
  for (var name in names)
    req.addParam(name);

  return con.request(req);
}

File: src/spice2x/api/resources/python/spiceapi/analogs.py
from .connection import Connection
from .request import Request


def analogs_read(con: Connection):
    res = con.request(Request("analogs", "read"))
    return res.get_data()


def analogs_write(con: Connection, analog_state_list):
    req = Request("analogs", "write")
    for state in analog_state_list:
        req.add_param(state)
    con.request(req)


def analogs_write_reset(con: Connection, analog_names=None):
    req = Request("analogs", "write_reset")

    # reset all analogs
    if not analog_names:
        con.request(req)
        return

    # reset specified analogs
    for analog_name in analog_names:
        req.add_param(analog_name)
    con.request(req)
